src/passes/Inlining.cpp

```diff
@@ -338,11 +338,25 @@ static Expression* doInlining(Module* module,
   //
   // Here the br wants to go to the very outermost block, to represent a
   // return from the inlined function's code, but it ends up captured by an
-  // internal block.
-  if (BranchUtils::hasBranchTarget(from->body, block->name)) {
-    auto existingNames = BranchUtils::getBranchTargets(from->body);
-    block->name = Names::getValidName(
-      block->name, [&](Name test) { return !existingNames.count(test); });
+  // internal block. We also need to be careful of the call's children:
+  //
+  //  (block $X             ;; a new block we add as the target of returns
+  //    (local.set $param
+  //      (call's first parameter
+  //        (br $X)         ;; nested br in call's first parameter
+  //      )
+  //    )
+  //
+  // (In this case we could use a second block and define the named block $X
+  // after the call's parameters, but that adds work for an extremely rare
+  // situation.)
+  if (BranchUtils::hasBranchTarget(from->body, block->name) ||
+      BranchUtils::BranchSeeker::has(call, block->name)) {
```

Comment:
What is the difference between `BranchUtils::hasBranchTarget` and `BranchUtils::BranchSeeker::has`?

Reply:
Got it, thanks. Does it save work to check for existence before gathering all the names below? It seems like gathering names would be about the same amount of work.

Reply:
Gathering them requires us to build up a set (allocation+hashing) so it is a little more work at least. Probably not a lot in the general case. But it is extremely rare to have a collision here.

Reply:
Makes sense 👍